startreedata/thirdeye
# fix(ui): TE-2056 improve clarity of error messages

#### Issue(s)

- https://startree.atlassian.net/browse/TE-2056

#### Description

- The goal of this PR is to improve error messaging across ThirdEye
- There are 3 levels of error messaging now
    - Error code mapped user-friendly error messages: here we have recreated the following map from the backend on the UI for more user-friendly error messaging based on different error codes https://github.com/startreedata/thirdeye/blob/6d893fed07a2283f734a4540d53ee90ae13f6c04/thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/ThirdEyeStatus.java#L18
    - Generic error messages: Instead of showing large not-so-user-friendly error message we will now show a more generic user friendly error message
    - The error message alert now has a "Show excpetions" section that is collapsed by default and on expanding shows the large backend sent exception for easier debugging
    - For RBAC related errors we will now show a toast message "You are not authorised to perform this action" for better clarity
- Also fixed all unit tests across the app


#### Screenshots
![image](https://github.com/startreedata/thirdeye/assets/67183907/9e9ae868-0928-47ee-b4b0-9be03e89f2b7)
![image](https://github.com/startreedata/thirdeye/assets/67183907/0f1a1102-ee47-4de9-817d-fdbeef7d3d4f)
![image](https://github.com/startreedata/thirdeye/assets/67183907/64c840d3-afdd-4842-94f9-59cd5f54328e)
![image](https://github.com/startreedata/thirdeye/assets/67183907/484bbb06-1d82-43fc-a5d4-bd91e8efda41)

#### How to test
- See if you can try to reproduce some error case scenarios and check out the error toast message text

List steps to reproduce the issue(s) and test the change, if required.
